mypy/fastparse.py

```diff
@@ -1111,15 +1114,32 @@ def visit_JoinedStr(self, n: ast3.JoinedStr) -> Expression:
     def visit_FormattedValue(self, n: ast3.FormattedValue) -> Expression:
         # A FormattedValue is a component of a JoinedStr, or it can exist
         # on its own. We translate them to individual '{}'.format(value)
-        # calls -- we don't bother with the conversion/format_spec fields.
-        exp = self.visit(n.value)
-        exp.set_line(n.lineno, n.col_offset)
-        format_string = StrExpr('{}')
+        # calls, passing along specified format specifiers and conversions.
+        val_exp = self.visit(n.value)
+        val_exp.set_line(n.lineno, n.col_offset)
+        conversion_str = '' if n.conversion is None or n.conversion < 0 else '!' + chr(n.conversion)
+        format_string = None
+        if n.format_spec is None:
+            format_string = StrExpr('{' + conversion_str + '}')
+        else:
+            # Note that format specifiers may contain expressions that need to be
+            # evaluated at runtime as well.
+            format_spec_exp = self.visit(n.format_spec)
+            empty_string = StrExpr('')
+            empty_string.set_line(n.lineno, n.col_offset)
+            join_method = MemberExpr(empty_string, 'join')
+            join_method.set_line(empty_string)
+            args = ListExpr([StrExpr('{' + conversion_str + ':'), format_spec_exp, StrExpr('}')])
+            args.set_line(join_method)
+            format_string = CallExpr(join_method,
+                                     [args],
+                                     [ARG_POS],
+                                     [None])
```

Comment:
I have an idea for another way to handle the nested computation thing: format lets you do `'{:{}}.format(x, y)` where `x` is the thing to format and `y` is the format specifier.